app/actions/ActionsProvider.scala

```diff
@@ -92,11 +92,32 @@ class ActionsProvider @Inject()(val authAction: AuthorisedAction,
       .andThen(EndOfYearFilterAction(taxYear, inYearUtil, appConfig))
       .andThen(UserPriorDataRefinerAction(taxYear, cisSessionService, errorHandler))
 
-  def endOfYearWithSessionData(taxYear: Int, contractor: String): ActionBuilder[UserSessionDataRequest, AnyContent] =
+  def userPriorDataFor(taxYear: Int, contractor: String, month: String): ActionBuilder[UserPriorDataRequest, AnyContent] = {
```

Comment:
It might be good if this method is moved up close to the method it overloads. It is also a good practice to keep the private methods at the bottom and the public ones towards the top.